### PR TITLE
Rename axes= to dims= in frontend

### PR DESCRIPTION
**Pull request by jacobhinkle**
Currently we accept `axes=` for some ops like `fd.ops.sum` and `dims=` for others like `fd.ops.squeeze`.

This is a small attempt to make the frontend arguments more consistent. This change renames the `axis=` kwarg to `dim=` and the same for `axes=` -> `dims=`.

I think we're free to set our own convention, but for reference:
- PyTorch uses `dim=` in most places and accepts either a single dim or multiple using that same argument name, where applicable.
- Numpy uses `axis=` and, like PyTorch, accepts a list where applicable.
- `jax.lax` uses `dimensions=`

### CODE REVIEW

**Comment by jacobhinkle** [comment]
!build

**Review by jjsjann123 — APPROVED** [review]
:clap:

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-0.1.5
```
Comment (jjsjann123) [inline version.txt:1]: Thanks for remembering to do this. :yum:

**Comment by jacobhinkle** [comment]
!build

File: version.txt (hunk 1 above)
Comment (jjsjann123) [reply version.txt:1]: We need to automate this part... or at least have a CI check to assert when we do break API... something maybe @xwang233 would be interested?!

Comment (jacobhinkle) [reply version.txt:1]: Oh that's a good idea.

Comment (xwang233) [reply version.txt:1]: I'll check pytorch's CI implementation on detecting BC breaking changes. 😉
